db/schema.go

```diff
@@ -128,31 +127,37 @@ func (db *db) patchSchema(ctx context.Context, txn datastore.Txn, patchString st
 		return err
 	}
 
-	var newDescriptionsByName map[string]client.CollectionDescription
+	var newSchemaByName map[string]client.SchemaDescription
 	decoder := json.NewDecoder(strings.NewReader(string(newDescriptionJson)))
 	decoder.DisallowUnknownFields()
-	err = decoder.Decode(&newDescriptionsByName)
+	err = decoder.Decode(&newSchemaByName)
 	if err != nil {
 		return err
 	}
 
 	newCollections := []client.CollectionDefinition{}
-	newSchemaByName := map[string]client.SchemaDescription{}
-	for _, desc := range newDescriptionsByName {
-		col, err := db.newCollection(desc, desc.Schema)
+	for _, schema := range newSchemaByName {
+		if schema.Name == "" {
+			return ErrSchemaNameEmpty
+		}
+
+		collectionDescription, ok := collectionsByName[schema.Name]
+		if !ok {
+			return NewErrAddCollectionWithPatch(schema.Name)
+		}
+
+		col, err := db.newCollection(collectionDescription, schema)
```

Comment:
thought: If we change the current `updateCollection` to `updateSchema`, we will still need an `updateCollection` function to handle the collection specific update to the new schema version. This part might be better off in that new `updateCollection` function.

Reply:
This will be handled in https://github.com/sourcenetwork/defradb/issues/1963 and https://github.com/sourcenetwork/defradb/issues/1963 - it was going to be done in https://github.com/sourcenetwork/defradb/pull/1965 but it proved to be a large undertaking to do it properly.